vpr/src/route/parallel_connection_router.cpp

```diff
@@ -406,6 +420,7 @@ void ParallelConnectionRouter::timing_driven_route_connection_from_heap_thread_f
         // Should we explore the neighbors of this node?
 
         if (inode == sink_node) {
+            heap_.setMinPrio(new_total_cost);
```

Comment:
Since this is very heap-specific, I recommend putting this within the heap class itself! That way we can always be sure that the value is set properly. I would recommend adding this to the `add_to_heap` method. That way if the target is ever added to the heap we can set the min prio.

Also, you are not ever resetting the min prio? Between connections, how is the prio reset to infinity? Does the reset method handle it?

Also also, this min-prio optimization may work for the deterministic version. Can you please create an IFDEF that will turn this feature ON/OFF. We would want to test the deterministic version with / without it. The IFDEF can be in the heap class itself if needed.

Reply:
Please make it OFF by default.

Reply:
All issues are resolved.
> Also, you are not ever resetting the min prio? Between connections, how is the prio reset to infinity? Does the reset method handle it?

It was handled by the reset method in MQ codebase.